irohad/main/irohad.cpp

```diff
@@ -139,6 +145,22 @@ int main(int argc, char *argv[]) {
       return EXIT_FAILURE;
     }
 
+    // check if genesis block already existing
+    auto genesis_block_exist = false;
+    irohad.storage->getBlockQuery()->getTopBlocks(1).subscribe(
```

Comment:
This code gets the top block (`getTopBlocks(1)`) instead of first block. Please use `getTopBlockHeight` instead, comparing it with 0. The variable would be better called, `ledger_empty`, because it can contain more that just a genesis block.

Reply:
@lebdron Hi,

I was trying to do as you said (use `getTopBlockHeight()`) but I wasnt able to find that method anywhere .. 
I even searched `develop` branch, but didnt find the method there. Can you please guide me on that ?

Reply:
It was added in #1305, that is blocking call, so the code will be simplier

Reply:
My bad ! I should have `git pull origin develop` before asking this. Thanks